src/web/app/user-join-page.component.ts

```diff
@@ -10,9 +12,56 @@ import { Component, OnInit } from '@angular/core';
 })
 export class UserJoinPageComponent implements OnInit {
 
-  constructor() { }
+  isLoading: boolean = true;
+  hasJoined: boolean = false;
+  errorMessage: string = '';
+  entityType: string = '';
+  key: string = '';
+  institute: string = '';
+  userId: string = '';
+
+  constructor(private route: ActivatedRoute, private router: Router, private httpRequestService: HttpRequestService) {}
 
   ngOnInit(): void {
+    this.route.queryParams.subscribe((queryParams: any) => {
+      this.entityType = queryParams.entitytype;
+      this.key = queryParams.key;
+      this.institute = queryParams.institute;
```

Comment:
the param name is not `institute` but `instructorinstitution`.

e.g. `http://localhost:4200/web/join?key=FD90816E3F11917C0303952E3E8384B83B90B8B3B65518567F02F71BDA37E0A98ED3961173771B21EAA24AF12D90E0A9&instructorinstitution=NUS&entitytype=instructor`